homeassistant/components/media_player/__init__.py

```diff
@@ -279,6 +280,30 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     await component.async_setup(config)
 
+    async def async_handle_volume_up_service(
+        media_player: MediaPlayerEntity, call: ServiceCall
+    ) -> None:
+        """Handle turning volume up."""
+        # pylint: disable-next=protected-access
+        max_volume = media_player._media_player_option_max_volume
+        if (
+            volume_level := media_player.volume_level
+        ) is not None and volume_level + media_player.volume_step > max_volume:
+            return
+        await media_player.async_volume_up(**remove_entity_service_fields(call))
```

Comment:
This function does a sanity check of the volume to prevent increasing the volume above the limit. As discussed in the architecture proposal, integrations still have the responsibility to ensure calls to the volume up service do not exceed the maximum.

To not get in the way of integration who need to do additional checks when handling volume up service calls, we should maybe move this to the class to allow integrations to override it.